fml/synchronization/sync_switch.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_SYNCHRONIZATION_SYNC_SWITCH_H_
+#define FLUTTER_FML_SYNCHRONIZATION_SYNC_SWITCH_H_
+
+#include <forward_list>
+#include <functional>
+#include <mutex>
+#include "flutter/fml/macros.h"
+
+namespace fml {
+
+/// A threadsafe structure that allows you to switch between 2 different
+/// execution paths.
+class SyncSwitch {
```

Comment:
This class can be tested using the `fml_unittests` harness.

Reply:
Added, found a nasty bug foo_({}) -> foo_([] {})